db/revision.go

```diff
@@ -234,15 +234,18 @@ func (db *DatabaseContext) getOldRevisionJSON(docid string, revid string) ([]byt
 //   delta=true && shared_bucket_access=false
 //      - old revision stored, with expiry rev_max_age_seconds
 // Ignores
-func (db *Database) backupRevisionJSON(docId, newRevId, oldRevId string, newBody Body, oldBody []byte) {
+func (db *Database) backupRevisionJSON(docId, newRevId, oldRevId string, newBody Body, oldBody []byte, newAtts AttachmentsMeta) {
 
 	if !db.DeltaSyncEnabled() || db.Options.DeltaSyncOptions.RevMaxAgeSeconds == 0 {
 		_ = db.setOldRevisionJSON(docId, oldRevId, oldBody, db.Options.OldRevExpirySeconds)
 		return
 	}
 
 	if db.UseXattrs() {
-		newBodyBytes, marshalErr := json.Marshal(newBody)
+		newBodyCopy := newBody.ShallowCopy()
```

Comment:
As discussed, instead of copying, can update _attachments in the doc, marshal, then restore previous _attachments (if present).

Reply:
Also, let's stick a comment explaining why we need to do this.